addons/mass_mailing_sms/controllers/main.py

```diff
@@ -42,19 +42,18 @@ def blacklist_number(self, mailing_id, trace_code, **post):
         if not check_res.get('trace'):
             return werkzeug.utils.redirect('/web')
         country_code = request.session.get('geoip', False) and request.session.geoip.get('country_code', False) if request.session.get('geoip') else None
-        trace = check_res['trace']
-        mailing_list_ids = trace.mass_mailing_id.contact_list_ids
-
         # parse and validate number
         sms_number = post.get('sms_number', '').strip(' ')
         sanitize_res = phone_validation.phone_sanitize_numbers([sms_number], country_code, None)[sms_number]
         tocheck_number = sanitize_res['sanitized'] or sms_number
 
+        trace = check_res['trace'].filtered(lambda r: r.sms_number == tocheck_number)
```

Comment:
As far as I can see, `trace` could still have `length > 1` after filtering.

```suggestion
        trace = check_res['trace'].filtered(lambda r: r.sms_number == tocheck_number)[:1]
```

Reply:
:+1: